src/editors/Flow/view/Core/Graph/GraphBase.js

```diff
@@ -607,18 +615,18 @@ export default class GraphBase {
    * @param {Boolean} status : True -> Running / False -> Not Running
    * @param {TreeContainerNode} parent : Flow to look for the node
    */
-  updateNodeStatus = (nodeName, status) => {
+  updateNodeStatus = (nodeName, status, parent) => {
     // is this a subflow node?
-    if (nodeName.indexOf("__") >= 0) {
-      const nodePath = nodeName.split("__");
-      const nodeParent = this.nodes.get(nodePath[0]);
+    const nodePath = nodeName.split("__");
 
-      if (nodeParent)
-        nodeParent.obj.status = [1, true, "true"].includes(status);
-    }
+    for (let i = 0; i < nodePath.length; i++) {
+      const nodeParent = this.getNodeParent(nodePath, i, parent);
 
-    const node = this.nodes.get(nodeName);
-    if (node) node.obj.status = [1, true, "true"].includes(status);
+      if (nodeParent) {
+        this.setNodeStatus(nodeParent, status);
+        parent = nodeParent;
+      }
+    }
```

Comment:
From what I understood (bare in mind I'm wrong more often than not :laughing:) this is the whole solution for your [ticket](https://movai.atlassian.net/browse/FP-2473).
I don't think adding this whole extra layer of caching which is prone to errors / bugs, is needed, at least at this point in time. @PedroCristovao-Movai what do you think?

Reply:
There are two tickets included here. I don't that is enough to fix the first one.

Reply:
I would like to have a solution that uses the architecture of flow (re-using what we already have), but from what I understood that way is too complex (Paulo tried and couldn't find a solution, in a short time). @manuelnogueiramov I don't mind if you try a different solution.

Reply:
For me, if it is working properly, and we don't have an alternative solution. I will accept the PR